core/src/modules/ops/source.ts

```diff
@@ -511,76 +514,119 @@ export namespace SourceOps {
    */
   export async function bootstrapUniqueProperty(
     source: Source,
-    key: string,
-    type: string,
-    mappedColumn: string,
-    id?: string,
-    local = false,
-    propertyOptions?: { [key: string]: any }
+    params: BootstrapUniquePropertyParams
   ) {
-    const property = Property.build({
-      id: id ?? ConfigWriter.generateId(key),
-      key,
-      type,
-      state: "ready",
-      unique: true,
-      sourceId: source.id,
-      isArray: false,
-    });
+    const {
+      mappedColumn,
+      id,
+      local = false,
+      propertyOptions,
+      sourceOptions,
+    } = params;
 
-    try {
-      // manually run the hooks we want
-      Property.generateId(property);
-      await Property.ensureUniqueKey(property);
-      await Property.ensureNonArrayAndUnique(property);
+    let { key, type } = params;
 
-      // danger zone!
-      await LoggedModel.logCreate(property);
-      // @ts-ignore
-      await property.save({ hooks: false });
+    const model =
+      !key || !type ? await GrouparooModel.findById(source.modelId) : undefined;
 
-      // build the default options
-      const { pluginConnection } = await source.getPlugin();
-      if (!local) {
-        let ruleOptions: SimplePropertyOptions = {};
+    let didGenerateKey = false;
+    const generateKey = (index?: number) =>
+      `${ConfigWriter.generateId(model.name)}_${ConfigWriter.generateId(
+        mappedColumn
+      )}${!index ? "" : "_" + index}`;
```

Comment:
I think there should be some tests locking in the behavior of this method when there is a conflict (I don't think I saw any of those added)